projects/igniteui-angular/src/lib/core/styles/components/grid-toolbar/_grid-toolbar-theme.scss

```diff
@@ -205,30 +206,39 @@
     }
 
     %igx-grid-toolbar--cosy {
-        height: map-get($grid-toolbar-height, 'cosy');
+        min-height: map-get($grid-toolbar-height, 'cosy');
         @include if-ltr() {
             padding: map-get($grid-toolbar-padding, 'cosy');
         }
 
         @include if-rtl() {
             padding: map-get($grid-toolbar-padding-rtl, 'cosy');
         }
+
+        %igx-button--outlined,
+        %igx-button--raised,
+        %igx-button--flat,
+        %igx-button--icon{
+            margin-#{$left}: rem(12);
+        }
     }
 
     %igx-grid-toolbar--compact {
-        height: map-get($grid-toolbar-height, 'compact');
+        min-height: map-get($grid-toolbar-height, 'compact');
         @include if-ltr() {
             padding: map-get($grid-toolbar-padding, 'compact');
         }
 
         @include if-rtl() {
             padding: map-get($grid-toolbar-padding-rtl, 'compact');
         }
-    }
 
-    %grid-caption {
-        display: flex;
-        align-items: center;
+        %igx-button--outlined,
+        %igx-button--raised,
+        %igx-button--flat,
+        %igx-button--icon{
+            margin-#{$left}: rem(8);
+        }
     }
 
     %igx-grid-toolbar__title {
```

Comment:
The title should probably not grow infinitely. It should probably be limited in width.